hack/lib/certmanager.bash

```diff
@@ -12,12 +12,27 @@ function uninstall_certmanager {
 
 function deploy_certmanager_operator {
   logger.info "Installing cert manager operator"
-  oc apply -f "${certmanager_resources_dir}"/subscription.yaml || return $?
 
-  logger.info "Waiting until cert manager operator is available"
+  openshift_version=$(oc version -o yaml | yq read - openshiftVersion)
+  deployment_namespace="cert-manager"
+  if printf '%s\n4.12\n' "${openshift_version}" | sort --version-sort -C; then
+      # OCP version is older as 4.12 and thus cert-manager-operator is only available as tech-preview in this version (cert-manager-operator GA'ed in OCP 4.12)
+      
+      echo "Running on OpenShift ${openshift_version} which supports cert-manager-operator only in tech-preview"
+
+      yq delete "${certmanager_resources_dir}"/subscription.yaml --doc 1 spec | \
+      yq write - --doc 2 spec.channel tech-preview | \
+      oc apply -f - || return $?
```

Comment:
If I run this locally, I get the resource changed, so I have to revert those changes before committing, can we use a temp file to change the version?

Reply:
You mean the subscription.yaml? On which yq version are you? The one I tested, requires the `--inplace` flag on write/delete to change the underlying resource

Reply:
```
$ yq delete hack/lib/certmanager_resources/subscription.yaml --doc 1 spec >/dev/null                                                                                                                                                                                                                                    $  git status                                                    
On branch install-cert-manager-from-stablev1-starting-4.12
Your branch is up to date with 'origin/install-cert-manager-from-stablev1-starting-4.12'.

nothing to commit, working tree clean
```

Reply:
I see, it's all good then